x-pack/plugins/security_solution/public/cloud_security_posture/components/vulnerabilities/vulnerabilities_insight.tsx

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+import { EuiFlexItem, type EuiFlexGroupProps } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { useVulnerabilitiesPreview } from '@kbn/cloud-security-posture/src/hooks/use_vulnerabilities_preview';
+import { buildEntityFlyoutPreviewQuery } from '@kbn/cloud-security-posture-common';
+import { getVulnerabilityStats, hasVulnerabilitiesData } from '@kbn/cloud-security-posture';
+import { InsightDistributionBar } from '../../../flyout/document_details/shared/components/insight_distribution_bar';
+
+/*
+ * Vulnerabilities insight in the alert/event flyout.
+ */
+export const VulnerabilitiesInsight = ({
+  hostName,
+  direction,
+  'data-test-subj': dataTestSubj,
+}: {
+  hostName: string;
+  direction?: EuiFlexGroupProps['direction'];
+  ['data-test-subj']?: string;
```

Comment:
extract these into a `VulnerabilitiesInsightProps` interface and docs each prop?